corehq/apps/cloudcare/views.py

```diff
@@ -105,27 +107,15 @@ class FormplayerMain(View):
     def dispatch(self, request, *args, **kwargs):
         return super(FormplayerMain, self).dispatch(request, *args, **kwargs)
 
-    def fetch_app(self, domain, app_id):
-        return _fetch_build(domain, self.request.couch_user.username, app_id)
+    def fetch_app_fn(self):
+        return get_latest_build_for_web_apps
 
-    def get_web_apps_available_to_user(self, domain, user):
-        app_access = get_application_access_for_domain(domain)
-        app_ids = get_app_ids_in_domain(domain)
-
-        apps = list(map(
-            lambda app_id: self.fetch_app(domain, app_id),
-            app_ids,
-        ))
-        apps = filter(None, apps)
-        apps = filter(lambda app: app.get('cloudcare_enabled') or self.preview, apps)
-        # Backwards-compatibility - mobile users haven't historically required this permission
-        if user.is_web_user() or user.can_access_any_web_apps(domain):
-            apps = filter(lambda app: user.can_access_web_app(domain, app.get('copy_of', app.get('_id'))), apps)
-        if toggles.WEB_APPS_PERMISSIONS_VIA_GROUPS.enabled(domain):
-            apps = filter(lambda app: app_access.user_can_access_app(user, app), apps)
+    def get_web_apps_for_user(self, domain, user):
+        apps = get_web_apps_available_to_user(
+            domain, user, is_preview=self.preview, fetch_app_fn=self.fetch_app_fn()
+        )
         apps = [_format_app_doc(app) for app in apps]
-        apps = sorted(apps, key=lambda app: app['name'].lower())
-        return apps
+        return sorted(apps, key=lambda app: app['name'].lower())
```

Comment:
Unrelated to the changes in this PR: is the result of this function always used in a context where it needs to be sorted (like because is displayed to a user)? If not, it would be good to move the sorting to where it's required since sorting is a relatively expensive operation.

Reply:
Yeah the two uses are [here](https://github.com/dimagi/commcare-hq/blob/6a6c72985c14082b77a75ef05998d39447faf597/corehq/apps/cloudcare/views.py#L176-L183), one adds `apps` into a context used to render a page, and the `get_option_apps` usage appears to be called from our javascript code, so both seem necessary. I specifically kept the sorting in this method and not in the new utility function for the reason you highlighted.